pkg/crc/config/viper_config.go

```diff
@@ -45,21 +44,43 @@ func (c *ViperStorage) Get(key string) interface{} {
 }
 
 func (c *ViperStorage) Set(key string, value interface{}) error {
-	c.viper.Set(key, value)
-	return atomicWrite(c.viper, c.configFile)
+	in, err := ioutil.ReadFile(c.configFile)
+	if err != nil {
+		return err
+	}
+	var cfg map[string]interface{}
+	if err := json.Unmarshal(in, &cfg); err != nil {
+		return err
+	}
+	cfg[key] = value
+	bin, err := json.MarshalIndent(cfg, "", "  ")
+	if err != nil {
+		return err
+	}
+	if err := atomicWrite(bin, c.configFile); err != nil {
+		return err
+	}
+	return c.viper.ReadInConfig()
```

Comment:
Adding more and more IO for each `Set`/`Unset`/... does not feel like they way to go to be honest :-/

Reply:
Yes.. I added as many tests as possible so that we can refactor that someday if we have better ideas.

Reply:
Oh I have better ideas, but this requires significant work, so not quite realistic now ;) (real IPC signaling rather than using files for config change notifications, better eventloop in crc so that we can delay writing of the config file, ...)